src/mappings/wLuna.ts

```diff
@@ -6,88 +6,102 @@ import {
 } from '../../generated/wLuna/wLuna'
 import { Approval, Burn, OwnershipTransferred, Transfer, Token, User } from '../../generated/schema'
 
+const LUNA = "LUNA"
+//const LUNA_ADDRESS = '0x320bc76961fb4e2a0e2e86d43d4b9d13b4985b8f'
+const ZERO_ADDRESS = '0x0000000000000000000000000000000000000000'
+
 export function handleApproval(event: ApprovalEvent): void {
-  let entity = new Approval(event.transaction.hash.toHex() + '-' + event.logIndex.toString())
+  let entity = new Approval(event.transaction.hash.toHex())
   entity.owner = event.params.owner
   entity.spender = event.params.spender
-  entity.value = event.params.value
+  entity.amount = event.params.value
   entity.save()
 }
 
-export function handleBurn(event: BurnEvent): void {
-  const id = 'LUNA'
-  let wLuna = Token.load(id)
-  if(!wLuna) {
-    wLuna = new Token(id)
-  }
-  wLuna.amount = wLuna.amount.minus(event.params.amount)
-  wLuna.totalBurned = wLuna.totalBurned.plus(event.params.amount)
+export function handleOwnershipTransferred(event: OwnershipTransferredEvent): void {
+  let entity = new OwnershipTransferred(event.transaction.hash.toHex())
+  entity.previousOwner = event.params.previousOwner
+  entity.newOwner = event.params.newOwner
+  entity.save()
+}
 
-  const userId = event.params._sender.toHex()
-  let user = User.load(userId)
-  if(!user) {
-    user = new User(userId)
+const getToken = (tokenName: string): Token => {
+  let token = Token.load(tokenName)
+  if (!token) {
+    token = new Token(tokenName)
   }
-  user.wLunaBalance = user.wLunaBalance.minus(event.params.amount)
-  user.txCount = user.txCount + 1
+  return token
+}
+
+export function handleBurn(event: BurnEvent): void {
+  const token = getToken(LUNA)
+  const burnAmount = event.params.amount
+  //token.amount = token.amount.minus(burnAmount)
+  token.totalBurned = token.totalBurned.plus(burnAmount)
```

Comment:
is `token.totalBurned` always 0 when doing `new Token("LUNA")`? Otherwise it could cause problems when adding the burnAmount

Reply:
it is, the constructor set that value to 0